### PR TITLE
Fix Travis Windows build

### LINKED ISSUE #3306: Travis Windows: Secret ENV vars break the build
The previous build issues are resolved. Now the UI tests are failing: 

https://travis-ci.org/rust-lang-nursery/rust-clippy/jobs/440797682#L746

I'm not sure what the problem is. The stderr diffs look the same on the first glance.

### PR DESCRIPTION
**Pull request by phansch**
Closes #3306

### CODE REVIEW

**Comment by mati865** [comment]
Strange, `cat` after `dos2unix` shows the line endings haven't changed (still are CRLF). `dos2unix` stdout is totally out of place and shows only a few lines.

**Comment by phansch** [comment]
I don't know what's going on with the cat :cat2: and dos2unix =(

Looking at the test output again, specifically the diffs: https://travis-ci.org/rust-lang-nursery/rust-clippy/jobs/451994929#L1102
They are indented by one space. This seems to come from [compiletest-rs/src/runtest.rs#L2502](https://github.com/laumann/compiletest-rs/blob/bc26d56a31703dcd6ea9246e6179b0a60c8cfede/src/runtest.rs#L2502), the `Both` case.

I guess I will try it with a patched compiletest-rs to print newline characters using `{:?}` instead of `{}`.

**Comment by phansch** [comment]
oh wait :man_facepalming:! Looking at the past logs again, it looks like dos2unix only formats the first directory that's globbed from `**/*`?

```
dos2unix: converting file tests/ui/author/for_loop.stderr to Unix format...
dos2unix: converting file tests/ui/author/matches.stderr to Unix format...
dos2unix: converting file tests/ui/author/call.rs to Unix format...
dos2unix: converting file tests/ui/author/for_loop.rs to Unix format...
dos2unix: converting file tests/ui/author/matches.rs to Unix format...
```

I'm going to give it a try with some sort of `find`/`xargs` loop

**Comment by phansch** [comment]
Marked as WIP because this will need a rebase tomorrow and Travis doesn't start right now either. I'm pretty convinced that the updated `find` command will make the Windows UI tests pass.

**Comment by mati865** [comment]
Well done, looks better:
```
failures:
    [ui] ui\author.rs
    [ui] ui\author\call.rs
    [ui] ui\author\for_loop.rs
    [ui] ui\trailing_zeros.rs
test result: FAILED. 219 passed; 4 failed; 1 ignored; 0 measured; 0 filtered out
```

**Comment by phansch** [comment]
The UI and UI-Toml tests pass now. The dogfood tests are failing now:

```
++++ pwd
+++ CLIPPY='/c/Users/travis/build/rust-lang-nursery/rust-clippy/target/debug/cargo-clippy clippy'
+++ /c/Users/travis/build/rust-lang-nursery/rust-clippy/target/debug/cargo-clippy clippy --all-targets --all-features -- -D clippy::all -D clippy::internal -Dclippy::pedantic
error: failed to run `rustc` to learn about target-specific information
Caused by:
  process didn't exit successfully: `C:\Users\travis\build\rust-lang-nursery\rust-clippy\target\debug\clippy-driver.exe rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro` (exit code: 3221225781)
```

I wonder what's up with `--crate-name ___`?

**Comment by flip1995** [comment]
In the appveyor config we add the binaries with https://github.com/rust-lang-nursery/rust-clippy/blob/3bb88775de5a6f8f5c5743dd11af6c820122ccc7/appveyor.yml#L27 to the `PATH`. The same should work with the `lib`s.

**Comment by mati865** [comment]
Lib won't work for Windows, all dynamic libs are placed in /bin.

**Comment by phansch** [comment]
That worked! I now also removed the windows build from `allow_failures`. Thanks for your help =)

File: ci/base-tests.sh [hunk 1]
```diff
@@ -27,6 +27,9 @@ cd clippy_dev && cargo test && cd ..
 # Perform various checks for lint registration
 ./util/dev update_lints --check
 
+echo $PATH
```
Comment (flip1995) [inline ci/base-tests.sh]: I think this debug output can also be removed

Comment (phansch) [reply ci/base-tests.sh]: Removed in latest rebase

**Comment by flip1995** [comment]
Nice, windows passes now! Let's bring this on it's way, but still keep appveyor until travis windows is (more) stable.

bors r+

**Comment by phansch** [comment]
huh, the windows build [failed without error](https://travis-ci.org/rust-lang-nursery/rust-clippy/jobs/453257566). I'm going to restart it.

**Comment by phansch** [comment]
bors retry

**Comment by phansch** [comment]
I guess the `allow_failure` wouldn't have worked either with that typo in there   :man_facepalming:

**Comment by phansch** [comment]
bors r=flip1995

**Comment by phansch** [comment]
bors retry

**Comment by mati865** [comment]
Probably bors is down.

bors ping

**Comment by phansch** [comment]
New day, another try

bors try

**Comment by phansch** [comment]
The failure should have been fixed afaict? :confused: 

bors r=flip1995

**Comment by phansch** [comment]
bors try

**Comment by phansch** [comment]
bors try

**Comment by phansch** [comment]
bors try

**Comment by phansch** [comment]
I believe this is a problem on Travis CI. I posted a comment on their support forum where some people have similar problems: https://travis-ci.community/t/choco-install-hangs-forever/307/14

We should not merge this until the problem is fixed, because using bors would also fail with other pull requests then.

**Comment by flip1995** [comment]
Always keep trying!

bors try

**Comment by phansch** [comment]
bors try

**Comment by phansch** [comment]
(We need to wait for bors to be added again to test this)

**Comment by flip1995** [comment]
@bors try

Just trying again since https://travis-ci.community/t/choco-install-hangs-forever/307/18 states that it works now on `travis.org`.

**Comment by phansch** [comment]
The try build still fails with a weird error: https://travis-ci.org/rust-lang/rust-clippy/jobs/472082461 :/

**Comment by flip1995** [comment]
This seems to be another Windows bug, since it is unrelated with the changes, that were made here: https://github.com/rust-lang/rust-clippy/pull/3577#issuecomment-449847720

Travis tries to load pty, but that doesn't exist on Windows. [StackOverflow](https://stackoverflow.com/a/35963637)
Let's postpone this further :/

**Comment by flip1995** [comment]
It seems like the windows build is working now, except https://travis-ci.com/rust-lang/rust-clippy/jobs/215431274#L1266, which should be fixable with https://github.com/rust-lang/rust-clippy/pull/3418#issuecomment-436988813

Let's try this.

**Comment by flip1995** [comment]
Well that's weird: https://travis-ci.com/rust-lang/rust-clippy/jobs/215447168#L2736-L2739

```
++++ ./target/debug/clippy-driver --print sysroot
+++ sysroot='C:\Users\travis\.rustup/toolchains/master'
++++ rustc --print sysroot
+++ test 'C:\Users\travis\.rustup/toolchains/master' = 'C:\Users\travis\.rustup\toolchains\master'
```

`clippy-driver --print sysroot` has a different output than `rustc --print sysroot`, even though `clippy-driver` just forwards the command IIUC.

**Comment by flip1995** [comment]
That (95477aa) fixed stuff. But now the tests for the `/tmp` sysroot fail (kind of expected). Disable / Replace them on windows builds?

**Comment by flip1995** [comment]
Windows build passes now, but takes 10 minutes longer than the Linux build. Let's try if `CARGO_INCREMENTAL=0` is still necessary

**Comment by flip1995** [comment]
Removing `CARGO_INCREMENTAL=0` slowed down the build by another 8 minutes... Let's run a `try` run to see whether it works with bors and whether the build time was a coincidence.

@bors try

**Comment by phansch** [comment]
We could maybe skip the dogfood tests on windows? I'm not sure but I guess that's where most of the time is spent in the windows build.

**Comment by flip1995** [comment]
Build time is up by 8 minutes without `CARGO_INCREMENTAL=0`. So let's leave this in.

Removing dogfood is a good idea, since the results should be the same on windows and linux/macos.

---
To improve the build time of every build: Maybe we could also cache RTIM, to not have to install every time?

**Comment by flip1995** [comment]
Without dogfood we now have about the same build times across the platforms. 

@bors try

---

I think the important question before merging this is:
Should we still allow failures on windows builds or should we yolo it? :smile:

**Comment by flip1995** [comment]
Caching RTIM only saves 2-5 minutes of build time, but stores caches of the size of ~40MB. I don't think this is worth it.

**Comment by flip1995** [comment]
This should be ready for merge now. Last question:

Should we
1. allow failures on the windows build :eyes: 
2. or yolo it? :rocket:

**Comment by phansch** [comment]
If it works, it works :man_shrugging: Thanks for pushing this through!

@bors r=me,flip1995

**Pull request merged**